### PR TITLE
#170947434 user inputs validation

### PR DESCRIPTION
**Pull request by izzett222**
#### What does this PR do?
add user inputs validation feature
#### Description of Task to be completed?
- install express-validator
- added the validation middleware
- added validation middleware tests
#### How should this be manually tested?
- clone this repo
- run npm install
- set up database (see readme)
- run npm run seedDb
- run npm run dev
- access the signup endpoint(/api/v1/auth/signup) through postman
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
#170947434

### CODE REVIEW

**Review by AbdoulNuru — APPROVED** [review]

**Review by hakoemmy — APPROVED** [review]

**Review by blaise82 — APPROVED** [review]

**Review by RusimbiPatrick — CHANGES_REQUESTED** [review]
In the case of error messages, The status code should be 422(unprocessable entity) instead of 400(bad request)
**Example**
Screen Shot 2020-02-12 at 13 38 21

**Comment by izzett222** [comment]
> In the case of error messages, The status code should be 422(unprocessable entity) instead of 400(bad request)
> **Example**
> Screen Shot 2020-02-12 at 13 38 21

I changed the status code to 422

**Review by RusimbiPatrick — APPROVED** [review]

**Review by RusimbiPatrick — APPROVED** [review]

File: src/routes/api/user.routes.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import express from 'express';
+import validation from '../../validation/user.validation';
+
+const router = express.Router();
+const {
+  checkFirstName, checkLastName, checkValidEmail, checkExistingEmail, checkPassword, validateResult
+} = validation;
+router.post('/signup', checkFirstName, checkLastName, checkValidEmail, checkExistingEmail, checkPassword, validateResult);
```
Comment (kcharles52) [inline src/routes/api/user.routes.js]:
```suggestion

router.post('/signup', checkFirstName, checkLastName, checkValidEmail, checkExistingEmail, checkPassword, validateResult);
```
Add space after constants

File: package.json [hunk 2]
```diff
@@ -4,9 +4,10 @@
   "description": "A Social platform for the creative at heart",
   "main": "index.js",
   "scripts": {
-    "start": "node build/index.js",
+    "start": "node --require @babel/polyfill build/index.js",
     "dev": "nodemon --exec babel-node src/index.js",
-    "test": "nyc --reporter=html --reporter=text mocha  --require @babel/register 'src/**/**test.js' --exit",
+    "test": "NODE_ENV=test npm run migrateTest && NODE_ENV=test nyc --reporter=html --reporter=text mocha  --require @babel/polyfill --require @babel/register 'src/**/**test.js'  --timeout 200000 --exit",
```
Comment (kcharles52) [inline package.json:9]:
```suggestion
    "test": "NODE_ENV=test npm run migrateTest &&  nyc --reporter=html --reporter=text mocha  --require @babel/polyfill --require @babel/register 'src/**/**test.js'  --timeout 200000 --exit",
```

**Review by kcharles52 — CHANGES_REQUESTED** [review]

**Pull request merged**